xbmc/interfaces/builtins/SkinBuiltins.cpp

```diff
@@ -100,6 +102,47 @@ static int SetAddon(const std::vector<std::string>& params)
   return 0;
 }
 
+/*! \brief Select and set a skin bool setting.
+ *  \param params The parameters.
+ *  \details params[0] = Names of skin settings.
+ */
+static int SelectBool(const std::vector<std::string>& params)
+{
+  std::vector<std::pair<std::string, std::string>> settings;
+
+  CGUIDialogSelect* pDlgSelect = (CGUIDialogSelect*)g_windowManager.GetWindow(WINDOW_DIALOG_SELECT);
+  pDlgSelect->Reset();
+  pDlgSelect->SetHeading(CVariant{g_localizeStrings.Get(424)});
+
+  for (unsigned int i = 0 ; i < params.size() ; i++)
+  {
+    std::vector<std::string> values = StringUtils::Split(params[i], '|');
+    std::string label = g_localizeStrings.Get(atoi(values[0].c_str()));
+    settings.push_back(std::make_pair(label,values[1].c_str()));
```

Comment:
Talking of c++11, push_back can be replaced by emplace_back, and there is no need to do c_str() on the values item.
`settings.emplace_back(std::make_pair(label, values[1]));`

And maybe its missing a `if (values.size() == 2)` because any lookup of `values` will fail if `values` is empty, or contains 1 element.